### ISSUE TITLE
Latest build (84ee483) crashes hub on startup

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#1055]
About time to sleep, just wanted to quickly say that my deployment it crashing like this if upgrading from the earlier to the later of these:

release | date | status
-|-|-
jupyterhub-0.8-84ee483 | Mon, 10 Dec 2018 10:41:10 +0000 | crashing
jupyterhub-0.8-ccc1e6b | Tue, 04 Dec 2018 16:56:30 +0000 | OK

/cc: @manics @minrk 

```
[E 2018-12-10 22:57:16.748 JupyterHub app:1958]
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/app.py", line 1955, in launch_instance_async
        await self.initialize(argv)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/app.py", line 1689, in initialize
        await self.init_spawners()
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/app.py", line 1478, in init_spawners
        spawner = user.spawners[name]
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/user.py", line 125, in __getitem__
        self[key] = self.spawner_factory(key)
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/user.py", line 221, in _new_spawner
        spawner = spawner_class(**spawn_kwargs)
      File "/usr/local/lib/python3.6/dist-packages/kubespawner/spawner.py", line 104, in __init__
        super().__init__(*args, **kwargs)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/config/configurable.py", line 84, in __init__
        self.config = config
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 585, in __set__
        self.set(obj, value)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 574, in set
        obj._notify_trait(self.name, old_value, new_value)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 1139, in _notify_trait
        type='change',
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 1176, in notify_change
        c(change)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 819, in compatible_observer
        return func(self, change)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/config/configurable.py", line 186, in _config_changed
        self._load_config(change.new, traits=traits, section_names=section_names)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/config/configurable.py", line 168, in _load_config
        warn(msg)
      File "/usr/lib/python3.6/contextlib.py", line 88, in __exit__
        next(self.gen)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 1131, in hold_trait_notifications
        self.notify_change(change)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/traitlets.py", line 1176, in notify_change
        c(change)
      File "/usr/local/lib/python3.6/dist-packages/kubespawner/spawner.py", line 318, in _deprecated_changed
        setattr(self.hub, change.name, change.value)
      File "/usr/local/lib/python3.6/dist-packages/traitlets/utils/bunch.py", line 15, in __getattr__
        raise AttributeError(key)
    AttributeError: value
```

**Comment by manics** [comment]
Similarly for new installations too, though it's a different error. After login on jupyterhub-0.8-84ee483:
```
[D 2018-12-11 10:07:53.159 JupyterHub user:190] Creating <class 'kubespawner.spawner.KubeSpawner'> for test:
[E 2018-12-11 10:07:53.170 JupyterHub web:1670] Uncaught exception GET /hub/user/test/ (10.0.2.15)
    HTTPServerRequest(protocol='http', host='localhost:31212', method='GET', uri='/hub/user/test/', version='HTTP/1.1', remote_ip='10.0.2.15')
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/web.py", line 1592, in _execute
        result = yield result
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/handlers/base.py", line 987, in get
        spawner = user.spawner
      File "/usr/local/lib/python3.6/dist-packages/jupyterhub/user.py", line 239, in __getattr__
        raise AttributeError(attr)
    AttributeError: spawner

[D 2018-12-11 10:07:53.174 JupyterHub base:880] No template for 500
[D 2018-12-11 10:07:53.189 JupyterHub user:190] Creating <class 'kubespawner.spawner.KubeSpawner'> for test:
[E 2018-12-11 10:07:53.199 JupyterHub log:150] {
      "X-Forwarded-Host": "localhost:31212",
      "X-Forwarded-Proto": "http",
      "X-Forwarded-Port": "31212",
      "X-Forwarded-For": "10.0.2.15",
      "Upgrade-Insecure-Requests": "1",
      "Cookie": "jupyterhub-hub-login=\"2|1:0|10:1544522873|20:jupyterhub-hub-login|44:NWM1YmJmOTRkODIyNGUwNmFkYmIwZGFkNGM2YWM5Mzc=|81afbaa1e3c6c41a29588dfc850d4bf02d46464e244db3dea47a03551ab52695\"; csrftoken=JAVtnae29zf5IcK8m3mXvmiCpQPE9mlM; _ga=GA1.1.1577166663.1531407956; _xsrf=2|beea854c|bc15329170b73ee7631bd8bee40b3368|1540385692; grafana_sess=3a07e091b0eb2a3f; sessionid_prod52=t40zq4sihqe8dul1fu3brkxmqcu25im7; sessionid_test60=yo255mf1dc88mxfnj95kofjxq66y82fa; sessionid=ijm43dc7f3nh8v9ygwkn1psi5h32e1d2; username-localhost-8888=\"2|1:0|10:1543597153|23:username-localhost-8888|44:NDE2NzA5YTc5ZDRlNDE4ZDhhMWY1N2ExNDkwNDRlYTk=|67178c290288aaef9d0376531ba5dc2d591bf5b0b743a182e37d5c2cf13a3eb3\"; jupyterhub-session-id=9b9ec2bfb03445d495f9ef1ad7c13aa3",
      "Connection": "close",
      "Dnt": "1",
      "Referer": "http://localhost:31212/hub/login",
      "Accept-Encoding": "gzip, deflate",
      "Accept-Language": "en-GB,en;q=0.7,en-US;q=0.3",
      "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:63.0) Gecko/20100101 Firefox/63.0",
      "Host": "localhost:31212"
    }
[E 2018-12-11 10:07:53.202 JupyterHub log:158] 500 GET /hub/user/test/ (test@10.0.2.15) 43.71ms
```
This si the same error I saw in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1053#issuecomment-445886727

**Comment by minrk** [comment]
I believe this is fixed by #1056.
